docs/userman/gui/extension/dynamic_element/tabular_data_props.md

```diff
@@ -0,0 +1,215 @@
+```
+
+The detailed explanation of the code is as follows:
+
+- We use the [useDispatch](https://docs.taipy.io/en/latest/refmans/reference_guiext/functions/useDispatch/) hook to
+  dispatch actions to the store and initiate backend communications.
+- Additionally, the [useModule](https://docs.taipy.io/en/latest/refmans/reference_guiext/functions/useModule/) hook
+  retrieves the page module, enabling correct execution of backend functions.
+- To request an update for every dynamic property of an element on initial render, we use the
+  [useDispatchRequestUpdateOnFirstRender](https://docs.taipy.io/en/latest/refmans/reference_guiext/functions/useDispatchRequestUpdateOnFirstRender/#function-usedispatchrequestupdateonfirstrender)
+  hook provided by the Taipy GUI Extension API. This hook takes five parameters:
+  - `dispatch`: The React dispatcher associated with the context.
+  - `id`: The identifier of the element.
+  - `context`: The execution context.
+  - `updateVars`: The content of the `updateVars` property.
+- We also dispatch the
+  [createRequestDataUpdateAction](https://docs.taipy.io/en/latest/refmans/reference_guiext/functions/createRequestUpdateAction/)
+  hook to create a request data update action, which updates the context by invoking the
+  [get_data](https://docs.taipy.io/en/latest/refmans/reference/pkg_taipy/pkg_gui/pkg_extension/ElementLibrary/#taipy.gui.extension.ElementLibrary.get_data)
+  method of the backend library. This invocation triggers an update of front-end elements holding the data.
+
+The [createRequestDataUpdateAction](https://docs.taipy.io/en/latest/refmans/reference_guiext/functions/createRequestUpdateAction/)
+hook accepts eight parameters:
+
+- `name`: The name of the variable containing the requested data, as received in the property.
+- `id`: The identifier of the visual element.
+- `context`: The execution context.
+- `columns`: The list of columns required by the element emitting this action.
+- `pageKey`: The unique identifier for the data received from this action.
+- `payload`: The payload, specific to the component type (e.g., table, chart).
+- `allData`: A flag indicating if all data is requested.
+- `library`: The name of the extension library.
+
+## Exporting the React component {data-source="gui:doc/extension/example_library/front-end/src/index.ts"}
+
+When the component is entirely defined, it must be exported by the JavaScript library.
+This is done by adding the export directive in the file `<project dir>/<package dir>/front-end/src/index.ts`.
+
+```js
+import GameTable from "./GameTable";
+
+export { GameTable };
+```
+
+## Using the element in the application {data-source="gui:doc/extension/table_chess_game.py"}
+
+In the example below, we use the `game_table` element to display a chess game board. The board is represented as a
+two-dimensional list of strings, where each string represents a chess piece. The board is displayed in a table format
+using the `game_table` element.
+
+```py
```

Comment:
Why would you want to repeat the code as is? when we have the link just above?

Just explain what data is, and the element definition.